commit 899e8e86192afce363b05b30f79590eef12b9724 (refs/changes/45/3345/4)
Author: Rob Johnston <rob.johnston@joyent.com>
Date:   2018-02-20T19:06:12+00:00 (1 year, 8 months ago)
    
    OS-6597 Changes for OS-6490 break sparc build
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/usr/src/lib/fm/topo/modules/common/smbios/smbios_enum.c b/usr/src/lib/fm/topo/modules/common/smbios/smbios_enum.c
index 097bc2c09a..de043d8ba8 100644
--- a/usr/src/lib/fm/topo/modules/common/smbios/smbios_enum.c
+++ b/usr/src/lib/fm/topo/modules/common/smbios/smbios_enum.c
@@ -17,7 +17,9 @@
 #include <fcntl.h>
 #include <fm/libtopo.h>
 #include <fm/topo_mod.h>
+#ifdef	__x86
 #include <sys/mc.h>
+#endif
 #include <sys/fm/protocol.h>
 #include <string.h>
 #include <unistd.h>
@@ -530,6 +532,7 @@ err:
 static boolean_t
 has_mc_driver()
 {
+#ifdef	__x86
 	int mc_fd;
 	mc_snapshot_info_t mcs;
 
@@ -542,6 +545,9 @@ has_mc_driver()
 	}
 	(void) close(mc_fd);
 	return (B_TRUE);
+#else
+	return (B_TRUE);
+#endif
 }
 
 /*ARGSUSED*/
